src/CoreFoundation/CFArray.cs

```diff
@@ -119,13 +91,13 @@ public static CFArray FromNativeObjects (params INativeObject[] values)
 
 		public IntPtr GetValue (nint index)
 		{
-			return CFArrayGetValueAtIndex (handle, index);
+			return CFArrayGetValueAtIndex (GetCheckedHandle (), index);
 		}
 
-		public static unsafe IntPtr Create (params IntPtr[] values)
+		internal static unsafe IntPtr Create (params IntPtr[] values)
```

Comment:
Maybe rename methods, so that the public functions are:

```csharp
public static CFArray Create (params IntPtr[] values);
public static CFArray Create (params INativeObject[] values);
```

and then for the internal ones that return IntPtr, something like this (which doesn't matter as much):

```csharp
static IntPtr CreateHandle (params IntPtr[] values);
static IntPtr CreateHandle (params INativeObject[] values);
```

that way we follow our naming guidelines.

Reply:
Yes, but only if/when they are made `public`. Right now they remains `internal` (at member instead of type-level)

Reply:
I'm sorry I wasn't clear, I was referring to renaming these methods to `Create`:

https://github.com/xamarin/xamarin-macios/blob/f26f6c37849b4a8f843f979aac217c8d4f02d6f0/src/CoreFoundation/CFArray.cs#L72-L80

Reply:
hmm... I should have made them `internal` for this first step. In part because I did not want, nor need, to modify too many other files.

The other part is that `NSArray` already expose both `FromIntPtrs` and `FromNativeObjects`... but I that does not mean I like their names :) We can certainly decide to renamed when obsoleting the existing `NSArray` versions.